makefile.shade

```diff
@@ -86,14 +86,35 @@ var FULL_VERSION = '${PRODUCT_VERSION + "-" + E("DOTNET_BUILD_VERSION")}'
         sourceFiles.Add(Path.Combine(libuvRoot, "src/unix/linux-inotify.c"));
         sourceFiles.Add(Path.Combine(libuvRoot, "src/unix/linux-syscalls.c"));
 
+        var sources = string.Join(" ", sourceFiles);
+
         var outputDir = Path.Combine(ROOT, "src/libuv/bin/linux");
         var outputPath = Path.Combine(outputDir, "libuv.so");
         Directory.CreateDirectory(outputDir);
 
-        var sources = string.Join(" ", sourceFiles);
+        var armOutputDir = Path.Combine(ROOT, "src/libuv/bin/linux-arm");
+        var armOutputPath = Path.Combine(armOutputDir, "libuv.so");
+        Directory.CreateDirectory(armOutputDir);
+
+        var arm64OutputDir = Path.Combine(ROOT, "src/libuv/bin/linux-arm64");
+        var arm64OutputPath = Path.Combine(arm64OutputDir, "libuv.so");
+        Directory.CreateDirectory(arm64OutputDir);
+
+        var armelOutputDir = Path.Combine(ROOT, "src/libuv/bin/linux-armel");
+        var armelOutputPath = Path.Combine(armelOutputDir, "libuv.so");
+        Directory.CreateDirectory(armelOutputDir);
 
         Exec(CLANG,
             string.Format("{0} -lm -pthread -ldl -lrt -fPIC -shared -o {1}  -I{2}/include -I{2}/src -Wall -Wextra -Wno-unused-parameter -Wstrict-aliasing -g --std=gnu89 -pedantic -D_GNU_SOURCE -D_BUILDING_UV_SHARED=1", sources, outputPath, libuvRoot));
+
+        Exec(CLANG,
+            string.Format("{0} -target armv7a-linux-gnueabihf -B/usr/arm-linux-gnueabihf/bin/ -B/usr/arm-linux-gnueabihf/lib/ -isystem /usr/arm-linux-gnueabihf/include -lm -pthread -ldl -lrt -fPIC -shared -o {1}  -I{2}/include -I{2}/src -Wall -Wextra -Wno-unused-parameter -Wstrict-aliasing -g --std=gnu89 -pedantic -D_GNU_SOURCE -D_BUILDING_UV_SHARED=1", sources, armOutputPath, libuvRoot));
+
+        Exec(CLANG,
+            string.Format("{0} -target armv8a-linux-gnueabihf -B/usr/arm-linux-gnueabihf/bin/ -B/usr/arm-linux-gnueabihf/lib/ -isystem /usr/arm-linux-gnueabihf/include -lm -pthread -ldl -lrt -fPIC -shared -o {1}  -I{2}/include -I{2}/src -Wall -Wextra -Wno-unused-parameter -Wstrict-aliasing -g --std=gnu89 -pedantic -D_GNU_SOURCE -D_BUILDING_UV_SHARED=1", sources, arm64OutputPath, libuvRoot));
```

Comment:
Here is what toolchain is being used in corefx for arm64: https://github.com/dotnet/corefx/blob/f053a6cb8cf24cc1ce41f3f216819ebc0af83408/cross/arm64/toolchain.cmake

It's a bit surprising it does not say "arm" in it and it probably needs to be installed separately from g(cc/++)-arm-linux-gnueabi(hf) because compilation failed when I tried it.